client/state/sites/reducer.js

```diff
@@ -204,6 +205,17 @@ export function items( state = {}, action ) {
 			}, state );
 		}
 
+		case SITE_DOMAINS_PRIMARY_SET_COMPLETED: {
+			const { siteId, primaryDomainName } = action;
+			return {
+				...state,
+				[ siteId ]: {
+					...state[ siteId ],
+					URL: `http://${ primaryDomainName }`
```

Comment:
Unfortunately, I can't promise anything - I have a full plate already and redux is very much _not_ my forte ;)
By the time I'd have finished this you'd probably be on your next support rotation 😜